### DISCUSSION TITLE
MT3DMS examples p8 and p10.

### DISCUSSION THREAD

**Discussion by AndreLuiz55** [modflowpy/flopy#1485] [Q&A] [answered]
Hello

I have a question about the p08 and p10 codes of the MT3DMS examples, both are showing the following error:

[Errno 2] No such file or directory: '..\\data\\mt3d_test\\mt3dms\\p08shead.dat'

[Errno 2] No such file or directory: '..\\data\\mt3d_test\\mt3dms\\p10shead.dat'

Where do I find this shead.dat file? I looked in the files created by MT3DMS and MODFLOW and I didn't find it.

Thanks in advance to anyone who can help.

**Comment by wpbonelli** [comment, marked as answer]
Hi @AndreLuiz55, `p08shead.dat` and `p10shead.dat` are located in `examples/data/mt3d_test/mt3dms` relative to the project root, see [here](https://github.com/modflowpy/flopy/tree/develop/examples/data/mt3d_test/mt3dms). Are you running the MT3DMS example notebook? Depending where you're running it from, you may need to modify the relative path. You can use `os.getcwd()` to check the working location as seen by the notebook process.

To avoid situations like this we could consider updating the example notebooks to use a utility function to locate example data instead of manually handling paths. The idea being to find `examples/data` automatically, providing the notebooks are run from somewhere inside the `flopy` project. (We already do this in the flopy tests, so it should be a minimum effort change.)

An alternative possibility (probably longer-term) could be to allow automatically downloading example data. I have seen other projects do this, [e.g. PyVista](https://docs.pyvista.org/examples/00-load/create-pointset.html#create-a-pointset). This could allow working with example models without cloning the project repo, e.g. something like:

```python
from flopy import examples

mf, mt, conc, cvt, mvt = examples.download_mt3dms(workspace="some/dir")
```

Above omits parameters for simplicity. It would probably take some work to figure out the right approach, but just an idea.

**Reply by AndreLuiz55** [reply]
Thank you very much, 

I will follow your recommendations.

**Reply by wpbonelli** [reply]
No problem at all. Feel free to reach out with any other issues.

I will follow up to let you know if we do implement either of the options described above.

**Reply by wpbonelli** [reply]
Hi @AndreLuiz55, were you able to get the MT3DMS examples working?

**Reply by AndreLuiz55** [reply]
Hello @w-bonelli , yes I did, thanks for the attention, I copied the files to my machine locally and indicated the path in the code. Then I will try to implement the tips you proposed. Thanks a lot for the help.

**Reply by wpbonelli** [reply]
Great, glad to hear it.

The options I mentioned are really intended for the flopy development team internally, sorry that wasn't clear. Just ideas to make things more convenient and avoid errors like the one you encountered.
